app/src/main/java/nerd/tuxmobil/fahrplan/congress/schedule/HorizontalSnapScrollView.kt

```diff
@@ -0,0 +1,239 @@
+
+        /**
+         * Calculates the number of columns to display at a time based on the physical dimensions and
+         * the screen density of the device and the column count of the schedule. Further limiting
+         * factors are a maximum column count to be displayed and a minimum column width.
+         */
+        @VisibleForTesting
+        fun calculateDisplayColumnCount(
+            availablePixels: Int,
+            columnsCount: Int,
+            maxColumnCountForLayout: Int,
+            densityScaleFactor: Float,
+            minColumnWidthDip: Int
+        ): Int {
+            val columnCountLimit = min(columnsCount, maxColumnCountForLayout)
+            if (columnCountLimit == 1) {
+                return 1
+            }
+            val availableDips = availablePixels.toFloat() / densityScaleFactor
+            val minWidthColumnCount = floor((availableDips / minColumnWidthDip).toDouble()).toInt()
+            val columnCount = min(minWidthColumnCount, columnCountLimit)
+            return max(1, columnCount)
+        }
+    }
+
+    private val gestureDetector: GestureDetector
+    private val logging: Logging = Logging.get()
+    private var xStart = 0
+    private var displayColumnCount: Int
+    private lateinit var roomNames: HorizontalScrollView
+
+    private var roomsCount = NOT_INITIALIZED
+
+    /**
+     * Sets the rooms count so it can be used to calculate
+     * the room column dimensions in the next layout phase.
+     */
+    fun setRoomsCount(@IntRange(from = 1) roomsCount: Int) {
+        this.roomsCount = roomsCount
+    }
+
+    private var columnWidth: Int
+    fun getColumnWidth(): Int {
+        check(roomsCount != NOT_INITIALIZED){ """The "roomsCount" field must be initialized before invoking "getColumnWidth".""" }
+        return columnWidth
+    }
+
+    // The currently displayed column index
+    var activeColumnIndex = 0
+        private set
+
+    init {
+        displayColumnCount = resources.getInteger(R.integer.max_cols)
+        columnWidth = 0
+        gestureDetector = GestureDetector(context, YScrollDetector())
+        setOnTouchListener(OnTouchListener())
+    }
+
+    inner class YScrollDetector : SimpleOnGestureListener() {
+        override fun onDown(event: MotionEvent): Boolean {
+            xStart = event.x.toInt()
+//            logging.d(LOG_TAG, """onDown xStart: ${xStart} getMeasuredWidth: ${getMeasuredWidth()}""");
+            activeColumnIndex = ((scrollX * displayColumnCount).toFloat() / measuredWidth).roundToInt()
+            return super.onDown(event)
+        }
+
+        override fun onFling(
+            event1: MotionEvent,
+            event2: MotionEvent,
+            velocityX: Float,
+            velocityY: Float
+        ): Boolean {
+            val normalizedVelocityX = velocityX / resources.displayMetrics.density
+            val columns = ceil((normalizedVelocityX / SWIPE_THRESHOLD_VELOCITY * 3).toDouble()).toInt()
```

Comment:
> `SWIPE_THRESHOLD_VELOCITY * 3`

The meaning of `3` should be clarified by declaring a constant with a descriptive name for it.

Reply:
should be fixed

Reply:
Do you know why the value is `3`?

Reply:
I have no idea. It seems to be some multiplication with which the amount of columns scrolled is multiplied to scroll further, hence the name, but I don't know why `3` was chosen.

Reply:
Maybe, `3` has been chosen as a value to **subjectively** satisfy **some amount of flinging** over rooms (when there is a lot) as discussed in #22.

Reply:
That's probably it 👍